src/test/components/elements/inputs/RegistrationEmailPreference.test.tsx

```diff
@@ -0,0 +1,139 @@
+import userEvent from "@testing-library/user-event";
+import { screen } from "@testing-library/react";
+import { UserRole } from "../../../../IsaacApiTypes";
+import { RegistrationEmailPreference } from "../../../../app/components/elements/inputs/RegistrationEmailPreference";
+import { renderTestEnvironment } from "../../../utils";
+
+describe("RegistrationEmailPreference", () => {
+  const mockSetEmailPreferences = jest.fn();
+
+  const setupTest = (role: UserRole | "ANONYMOUS", props = {}) => {
```

Comment:
Given how many times I keep seeing this `UserRole | "ANONYMOUS"`, I think we should probably have a type for it. I'm not sure what the name should be. Or if we also need to change the name of `UserRole`. But something should be done. Not necessarily as part of this PR, but it's just something to think about.

Reply:
I've added a new type for this, specifically for use in testing.  `UserRole` is used across the app, but this `UserRole | "ANONYMOUS"` is testing specific, so we now have `TestUserRole` stored in src/test/utils.tsx